File: Algorithms/C++/1. ContainsDuplicate/ContainsDuplicate.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
    public:
        bool hasDuplicate(vector<int>& nums) {
            unordered_set<int> numSet;

            for(const int& num : nums){
                if(numSet.find(num) != numSet.end()){
                    return true;
                }

                numSet.insert(num);
            }
            return false;
        }
    };

File: Algorithms/Python3/217.ContainsDuplicate/ContainsDuplicate.py
class Solution:
    def hasDuplicate(self, nums: list[int]) -> bool:
        
        checkSuccess = set()

        for num in nums:
            if num in checkSuccess:
                return True
            checkSuccess.add(num)
            print(checkSuccess)
        
        return False
    
test = Solution()
test.hasDuplicate([1,2,3,4,3])
            